### ISSUE TITLE
Language selection not working.

### ISSUE THREAD

**Issue by micove** [PCSX2/pcsx2#416]
Seems the i18n code broke between d4793cc (works) and 5933db3 (dead)

![pcsx2_interfacelanguage](https://cloud.githubusercontent.com/assets/1458001/5608001/a39dc174-9440-11e4-8789-a6e87029168a.png)

Everything works fine until I click "Change Language".  The new window opens up and changes most of the windows to vietnamese (?) automatically in Linux.

If I close the program everything reverts and works fine until I click "Change Language" again.

**Comment by avih** [comment]
I'm guessing 2b567e51ff1a011fb89bf0901d6507944339753f broke it, though it only display the same language panel which the first time wizard display, so it can only mean that the panel is not as isolated as I assumed it is.

How did it behave before it got broken?

**Comment by micove** [comment]
Everything was in English before. This only triggers if I click the "Change Language" option and the window popups. As soon at it pops up the image shown appears and w/o me doing anything it goes Asian. If I never click the "Change Language" option it never triggers but I have not tried too much to trigger it since I found it just as I was about to leave my Linux machine.

To be more specific PCSX2 specific text is still in English. Mainly the system specific things like when the window to select an ISO or a different file from the system those appear in vietnamese. Like the "home"/"Desktop" folders transform into an whatever home/desktop is in vietnamese. The cancel button, apply button , etc too.

**Comment by avih** [comment]
You did not answer how it behaved before it got broken. "Everything was is English" is not a good enough description. Please provide something like this (and tell us which revision you're using): 

1. You clicked "Change language".
2. I saw this and that on screen.
3. I did foo and bar.
4. The result was buz.

**Comment by micove** [comment]
I did answer. Everything worked. The language did not change. The new window did not exist back then not sure what I'm supposed to report. Everything stayed in english.

The best I can say is the old window telling me to restart PCSX2 appeared and nothing happened.

**Comment by avih** [comment]
And after you restarted PCSX2?

**Comment by micove** [comment]
As I said with the old revision ... everything worked. The 1st configuration wizard started and no Vietnamese ever. I used that button a lot since I used it as my defacto "trigger 1st configuration wizard w/o having to delete my config folder option" hehe. So I'm pretty sure it worked as expected in that revision.

With the new revision after it opens the window the language changes. The only way to reset it is to restart PCSX2. It won't trigger unless I open the new window.

**Comment by avih** [comment]
What happens if (at the change language dialog) you choose some language, click Apply, then choose English again and click Apply again?

Regardless, you can trigger the first time wizard without deleting your configs, but this is off topic for this bug.

**Comment by micove** [comment]
Im the new one I changed languages twice and the Vietnamese remained. The first time it appeared the 1st thing I did was change the language out of english into english since I may have done something to break it but there is no po files for Vietnamese and the affected text is not in the po files.

In the old one I did not test it today but I have done it before and it just changed the Language in the GUI using as expected. I literally mean everything appeared to work since if the bug is old it was ultra latent since I had not triggered it before today but I had tested almost everything to safely say everything appeared to work but I can't say it did not exist since it could have been a latent bug that did not triggered until the new code was added.

**Comment by avih** [comment]
> In the old one I did not test it today but I have done it before and it just changed the Language in the GUI using as expected

I'd appreciate if you can confirm it still behaves the same with a recent version but before it broke (probably d4793cc6b6d5d66fa1e4856801c6289f3e6fc9d5 since you say it works).

Regardless though, my HDD died today, so it's going to take a while till I setup a new development machine.

**Comment by micove** [comment]
I will boot the linux box and retest  d4793cc since it should be trivial.
I will also try to compile the revision right before the one you suggested as the cause but this will take like 20 min.

**Comment by avih** [comment]
Appreciated.

**Comment by micove** [comment]
Okay, I'm compiling 576513a which is before the one you mentioned. I will just report the results when it slowwwwwwwlyyyyyyy compiles hehe.

Using d4793cc, in the 1st configuration wizard I selected arabic, spanish, german , english and a lot of languages in random order and could never trigger the Vietnamese issue. Everything seems to display in the correct language. I checked and my Desktop and Home folder are still in English when trying to select an ISO.

**Comment by avih** [comment]
So you're saying that you could select, for instance, Arabic at the first time wizard, continue, and then when PCSX2 opens it's in Arabic?

And that changing the language from the new smaller Change Language dialog is completely not letting you change language (or never applies the new language which you choose)?

**Comment by micove** [comment]
Using the old version with Spanish at the 1st time configuration:
![screenshot-pcsx2 1 3 0-0 git jan 4 2015](https://cloud.githubusercontent.com/assets/1458001/5608878/c71ff784-945e-11e4-9b21-06e867dad3ff.png)

Using the new one while selecting spanish:
![crop](https://cloud.githubusercontent.com/assets/1458001/5608921/be7c79bc-945f-11e4-832d-7ec2f19434f9.jpg)

**Comment by avih** [comment]
And what happens if you Click OK and then restart PCSX2?

Does it show in Spanish? I.e. is the issue only at the language at which the smaller dialog is displayed? or is it broken and just doesn't apply the language you select?

**Comment by avih** [comment]
Please help me understand if the issue is only the language at which the new dialog is displayed, or the issue is that it doesn't change your language when you choose a new one.

**Comment by micove** [comment]
Okay let me spell it out step by step as I do it.

1. Open pcsx2 with english as default
2. Everything is in english
3) go to the iso file everything is in english, no issues that I can find
4) Click "Change Language"
5) W/o doing anything else the Thoi , Apdung appears as the window appears.
6) Click Thoi (cancel) system locale stays in Vietnamese while the GUI stays with the english locale
7) Click "Change Language" and click OK this time
8) Vietnamese disappears (I was wrong before since apparently I hit cancel, sorry about that)
9) Click "Change Language", spanish and click OK again no Vietnamese again.
10)  Click "Change Language", english and click apply and then cancel. And Vietnamese goes away.

So if you hit OK or Ap Dung (hehe) it resets. Hitting Thoi it stays Vietnamese.

**Comment by micove** [comment]
Update:

Using 576513a which is the commit before the zombie mode commit did not produce the Vietnamese thing the new window appears and does the job as documented.

**Comment by avih** [comment]
Huh? So you're sating that 2b567e51ff1a011fb89bf0901d6507944339753f works, but 10040f82e050b3d64271e8c073850afd0f1e70f9 is broken? Are you 100% sure on that?

**Comment by micove** [comment]
No I'm saying

576513a  works
d4793cc works
 5933db3 (dead)

2b567e5 is the commit after the one I just tested so I can't say it works. I can only say the one before it works.
10040f8  is two commits after the one I just tested which is the same as the reasoning above.

**Comment by micove** [comment]
Those two are the only 2 commits left that could have cause it might as well test the 1st one
2b567e51ff1a011fb89bf0901d6507944339753f

**Comment by avih** [comment]
These are (99%) the only relevant changes:

First related change. I suspect this broke it.
- c4bd50c04c55470f6036c9ad8da2ec5268eadb2d UI: Change Language: use a dialog instead of forcing first-time-wizard

Later, change which practically does nothing but you claimed (twice) that it's working:
- 576513a6d0493327db366c7d20d525756202542c kb shortcuts: fixed file name PCSX2_keys.ini to PCSX2_keys.ini.default

Later, change which might affect languages but probably doesn't:
- 10040f82e050b3d64271e8c073850afd0f1e70f9 UI: change language: more standard dialog and better Apply button …

Which of those is broken and which is working? is there any version which displays the new language dialog correctly?

**Comment by micove** [comment]
As I mentioned before. I tested 576513a which works. I decided to test this one since it was the 1st commit before 2b567e5  whcih was the one you suspected it broke stuff. If the one before 2b567e5  works and 2b567e5  worked it was clear that that was the commit that introduced the regression.

The only ones untested that could do be related are:
2b567e5 
10040f82e050b3d64271e8c073850afd0f1e70f9

**Comment by avih** [comment]
> I decided to test this one since it was the 1st commit before 2b567e5 whcih was the one you suspected it broke stuff

No, I accidentally pasted this number, and edited it 2 hours ago (see the second comment).

The only relevant changes (with 99% certainty) are two:
1. The first one which added the dialog instead of the first time wizard.
2. The second which slightly changed this dialog.

I suspect it's the first of those which broke it, but you say that a revision between those two is working. Which suggests that it's the 2nd one which broke it.

Please see the 3 revisions mentioned just above your comment and let me know which of them is working. Please don't assume - try each of them and let me know which works.

Thanks

**Comment by micove** [comment]
Oh, hehe never saw the edit.

I already tested 576513a and it works.
I'm compiling 2b567e5  already.

So I will start compiling 
10040f8
and then
c4bd50c

**Comment by avih** [comment]
While you're compiling, I'd appreciate some clarifications on those:

- What's your default system locale regardless of PCSX2?


> 1. Open pcsx2 with english as default

OK.

> 5\ W/o doing anything else the Thoi , Apdung appears as the window appears.

OK, so is this a bug because the OK/Apply/Cancel buttons appear in Vietnamese, but you expect them to appear in English instead?

> 6\ Click Thoi (cancel) system locale stays in Vietnamese while the GUI stays with the english locale

What does that mean? what shows in English and what shows in Vietnamese after you close the dialog? And what happens when at this state you close and re-open PCSX2?

> 7\ Click "Change Language" and click OK this time
> 8\ Vietnamese disappears (I was wrong before since apparently I hit cancel, sorry about that)

I don't understand how are 7/8 different than 6. In both cases the selected language was correct, and then hitting either ok or cancel didn't change PCSX2? or did it change in different ways?

> So if you hit OK or Ap Dung (hehe) it resets. Hitting Thoi it stays Vietnamese.

What does "reset" mean? PCSX2 changes to English? And stays English also after you restart PCSX2?

Also, like the dialog says, the language will be inconsistent after you click OK/Apply and until you restart PCSX2. This is expected, so please don't experiment and don't report anything which happens after you click OK/Apply and before you restart PCSX2.

If I'm processing all the data I've seen so far, it seems to me that:
1. Changing the language at the dialog and then restarting PCSX2 - is working correctly - and PCSX2 starts with the newly chosen language.
2. The OK/Apply/Cancel appear in Some language other than what you expect, but other than that, it works.

What I still don't know:
- You're in english after pcsx2 is restarted.
- You click change language -> the OK/Apply/Cancel appear in some unexpected language.
- you now click cancel, or press escape:
  - did PCSX2 now change language unexpectedly?
  - After restarting PCSX2, is it now in unexpected language?

Is the whole issue that the OK/Apply/Cancel appear in unexpected language, but if you imagined that they displayed in the correct language - then everything is working just fine?

**Comment by micove** [comment]
>What's your default system locale regardless of PCSX2?
English UTF8

>OK, so is this a bug because the OK/Apply/Cancel buttons appear in Vietnamese, but you expect them to appear in English instead?

They always appeared in English before and as in the picture OK, Apply , Filesystem, Search, etc appear in Vietnamese.

>What does that mean? what shows in English and what shows in Vietnamese after you close the dialog? And what happens when at this state you close and re-open PCSX2?

Once again the big picture shows what stays in english (well spanish in this case). The buttons apply, cancel, and the system related stuff in the file selection dialogue.

>What does "reset" mean? PCSX2 changes to English? And stays English also after you restart PCSX2?

Whenever you open the "Change Language" at that point the system locale apparently changes to Vietnamese hence why always those two button appear in Vietnamese. If I select any language and hit OK/Apply (ap tung) then the Vietnamese reset or goes away and the selected locale takes it place. The window still says Thoi/Ap tung. But when I close it to open the file dialog the Vietnamese disappears.

If I open "Change Language" and hit cancel/Thoi instead of selecting to change a new language. The file dialogue shows the Vietnamese text.

I'm not complaining about the inconsistencies the only issue is that opening the "Change Language" always changes the locale to Vietnamese so if you open that window you always have to click OK/Apply with any language from the list unless you want to have Vietnamese. Since there is no Vietnamese po only text that they system can translate appear Vietnamese. I have not idea why it's vietnamese and not Thai or something else. I don't know Vietnamese or know anyone from there.


>The OK/Apply/Cancel appear in Some language other than what you expect, but other than that, it works.

Basically yes, as I mentioned opening the window seems to change the locale automatically to Vietnamese. So yeah if that behaviour is stopped everything works perfectly. More than Ok/Apply/Cancel is changed. The picture shows more.

**Comment by micove** [comment]
Amusingly enough:

2b567e5
10040f8

seem to work. I can't trigger the Vietnamese.

**Comment by avih** [comment]
Nothing touched languages as far as I know after 10040f8, so I'm really clueless on what's going on. Something is very inconsistent in your reports, and I'm unable to follow what is wrong and where.

The only thing I understand is that in some cases clicking the change language menu item shows a dialog with OK/Apply/Cancel buttons in unexpected language.

I don't understand when it happens, or what happens after that, or which revisions work and which are broken.

I hope someone else will be able to reproduce this and provide some more detailed info, or else I'll assume it's only an issue on your system which doesn't affect other people.

**Comment by micove** [comment]
I don't see any inconsistencies.

There is one revision that the broken behavior happens and all the ones the other ones I have tested before that one seem to work. The issue is not in some cases it ALWAYS happens in the broken revision.

I will just bisect this until I find a cause then.

It repeats every time I reopen that window. Since PCSX2 does not provide Vietnamese po/mo files then it uses the one already installed in my system which is why only stuff like Desktop/File System/Search, etc gets translated to Vietnamese.

**Comment by avih** [comment]
Sorry, unable to follow you.

If you can produce the issue consistently, then please identify explicitly the last revision which is working OK and the first revision which is broken (which should be one revision after the one which is working OK).

Until we have this info 100% solid, I can't continue the analysis.

**Comment by micove** [comment]
Yes that is what I just said I will do. I will just bisect this going commit by commit since the range is getting smaller.

**Comment by avih** [comment]
To summarize what we need:
1. Which is the first broken revision? (this is a must - without it we can't continue to 2).
2. Other than the buttons SHOWING in unexpected language - how does it BEHAVE unexpectedly? (and please describe it as "i expected X but instead Y happened").

**Comment by micove** [comment]
Not sure why the confusion since it's a rather simple issue with an apparently complex cause:

Whenever I open the "Change Language" window the window seem to trigger something that automatically sets the locale of the program to an unexpected language in my case Vietnamese.  This always happens. That is it. The rest of the behavior fits this explanation even if this is not what actually happens but the behavior matches what I just said.

Since PCSX2 does not provide vietnamese mo files then most of the text remains in the last locale used. While using the system's vietnamese mo files. That is why System related text seems to only be affected. If the bug had picked a language like automatically change to Japanese I would expect all the program to be Japanaese. 

This theory might be totally wrong but it explains the observed behavior perfectly.

Now that the window is opened and the locale automatically (allegedly)  set to Vietnamese. Then there are two thing that can happen.
- You hit cancel therefore you stays with the silently enabled Vietnamese locale.
- You select a new language which just changes back to the selected language. It basically goes 
ENGLISH -> VIETNAMESE (unknown reason) -> New language. Then English -> Vietnamese part always triggers that is why if you cancel the chain ends there.

Does this explanation makes it better to understand? I'm just trying to rephrase it ti make it easier to understand not trying to be rude or anything. :\.

IF the silent/automatic change of language that is trigger on the click event is solved then the issue is gone.

>However, issues which happen after opening the dialog and right afterwards (without doing anything else) hitting escape (or pressing Cancel) ARE valid issues.

Yes this is the only issue I'm reporting. That opening the window triggers the language to be changed automatically to something else w/o my consent or input. There is no other issue everything else works as expected. Since hitting cancel would just make it stay in the undesired language as expected. And hitting ok/apply changes the language to the selected one as expected. i never even intended to change the language I just wanted to see the new window and all this happened.


Update:

Ok my next guess was PR #403 
so I tested 798fa11c57d5ebf26d0f88ca84c01180ea7fa2e0 and it works.
To make it work I had to cherry pick your commit
gzip ISO: fix compile issues on linux 5933db386a886a2ee39f41cc84409c31b6110f50.

So only 3 commit untested.
cdd6598e65a28cd7718d8fae4c4777be24c225e9 is not used

Therefore this is the cause:
aea421cc8e0e45ac4d4eb55ea56d907aa2013ad1
a11286bbc54a32e0a73163689243b7f3869b7c23

Both are needed and it's equivalent to the buggy revision I first reported. 
5933db386a886a2ee39f41cc84409c31b6110f50

**Comment by micove** [comment]
I'm going to backport the PR and see if adding it after the old commits breaks the stuff too.

**Comment by avih** [comment]
> 1. Which is the first broken revision? (this is a must - without it we can't continue to 2) Not A version which is broken, but rather THE FIRST version which is broken and which the previous version works..

Just do this.
Don't assume, don't guess, don't cherry pick, don't PR.

Just find me the first revision which you consider broken, and which the prior revision is working OK, and we'll take it from there.

**Comment by micove** [comment]
I told you that is the PR.

Think a second here please.

W/o cherry picking that commit it does not build.

What I just did was revert the PR request and it works. I can't do anything else.

**Comment by avih** [comment]
You can go on with this forever, and I refuse to provide any more help until you tell me exactly this:
> 1. Which is the first broken revision? (this is a must - without it we can't continue to 2) Not A version which is broken, but rather THE FIRST version which is broken and which the previous version works..

**Comment by micove** [comment]
Ok I just told you. I tried to do it without the cherry pick but it fails to build.

That cherry pick fixes the failure to build. It won't compile.

**Comment by micove** [comment]
The title is 
"gzip ISO: fix compile issues on linux"

**Comment by avih** [comment]
So fix your compilation issues and then tell me what I need to hear.

**Comment by micove** [comment]
I fixed with the cherry pick.

**Comment by avih** [comment]
So tell me the last revision which you compiled successfully and works, and the first revision which you compiled successfully and which is broken.

**Comment by micove** [comment]
After fixing the build failure.

The one before the PR works and after the PR fails.

**Comment by avih** [comment]
Not before and after. revision hashes.

**Comment by micove** [comment]
798fa11c57d5ebf26d0f88ca84c01180ea7fa2e0 works
aea421cc8e0e45ac4d4eb55ea56d907aa2013ad1 fails

**Comment by avih** [comment]
Thank you.
Now cherry pick "gzip ISO: fix compile issues on linux" and bisect further into the rvisions which you couldn't compile, and tell me again which is the last one which works and which is the first which is broken.

**Comment by micove** [comment]
I had to cherry pick it. I'm telling you reverting PR 403 and it works.

**Comment by avih** [comment]
Even if I might understand, I can't be sure anymore. Please describe exactly:

1. I checkout A
2. I cherry-pick B
-- it works.

3. Then i checkout C (which MUST be A+1)
4. Then I cherry-pick D
-- it's broken

**Comment by gregory38** [comment]
I will give it a look

Note it could also be a wx3/wx2 discrepancy

**Comment by micove** [comment]
Sigh... okay

Here is the build logs

I'm just compiling as asked w/o cherry picked as I did originally.

```
[ 42%] Building CXX object pcsx2/CMakeFiles/pcsx2.dir/CDVD/CompressedFileReader.cpp.o
cd /tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu/pcsx2 && /usr/lib/ccache/c++   -DDOC_DIR_COMPILATION=/usr/share/doc/pcsx2 -DGAMEINDEX_DIR_COMPILATION=/usr/share/games/pcsx2 -DPLUGIN_DIR_COMPILATION=/usr/lib/i386-linux-gnu/pcsx2 -DWXUSINGDLL -DWX_PRECOMP -DXDG_STD -D_ARCH_32=1 -D_FILE_OFFSET_BITS=64 -D_M_X86=1 -D_M_X86_32=1 -D__WXGTK__ -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2  -msse -msse2 -march=i686 -pipe -std=c++0x -fvisibility=hidden -pthread -Wall -Wno-attributes -Wno-missing-field-initializers -Wno-unused-function -Wno-unused-parameter -Wno-unused-variable -Wno-unused-value  -Wstrict-aliasing -Wstrict-overflow=4  -D_FORTIFY_SOURCE=2  -Wformat -Wformat-security -DNDEBUG  -Wno-invalid-offsetof -pthread -I/usr/include/libdrm -I/usr/include/soundtouch -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/i386-linux-gnu/glib-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/lib/i386-linux-gnu/gtk-2.0/include -I/usr/include/SDL -isystem /usr/lib/i386-linux-gnu/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/common/include -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/common/include/Utilities -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/common/include/x86emitter -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu/common/include -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/gui -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/x86 -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu/pcsx2/gui -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/.    -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -fno-guess-branch-probability -fno-dse -fno-tree-dse -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -falign-functions -falign-jumps -falign-labels -falign-loops -fcaller-saves -fcprop-registers -fcrossjumping -fcse-follow-jumps -fcse-skip-blocks -fdefer-pop -fdelete-null-pointer-checks -fgcse -fgcse-lm -fif-conversion -fif-conversion2 -fmerge-constants -foptimize-sibling-calls -fpeephole2 -fregmove -freorder-blocks -freorder-functions -frerun-cse-after-loop -fsched-interblock -fsched-spec -fstrict-overflow -fthread-jumps -ftree-ccp -ftree-ch -ftree-copyrename -ftree-dce -ftree-dominator-opts -ftree-fre -ftree-lrs -ftree-pre -ftree-sra -ftree-ter -ftree-vrp -funit-at-a-time  -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -DWX_PRECOMP -fno-guess-branch-probability -fno-dse -fno-tree-dse -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -DWX_PRECOMP -falign-functions -falign-jumps -falign-labels -falign-loops -fcaller-saves -fcprop-registers -fcrossjumping -fcse-follow-jumps -fcse-skip-blocks -fdefer-pop -fdelete-null-pointer-checks -fgcse -fgcse-lm -fif-conversion -fif-conversion2 -fmerge-constants -foptimize-sibling-calls -fpeephole2 -fregmove -freorder-blocks -freorder-functions -frerun-cse-after-loop -fsched-interblock -fsched-spec -fstrict-overflow -fthread-jumps -ftree-ccp -ftree-ch -ftree-copyrename -ftree-dce -ftree-dominator-opts -ftree-fre -ftree-lrs -ftree-pre -ftree-sra -ftree-ter -ftree-vrp -funit-at-a-time -DXDG_STD -o CMakeFiles/pcsx2.dir/CDVD/CompressedFileReader.cpp.o -c /tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp: In function 'wxString ApplyTemplate(const wxString&, const wxDirName&, const wxString&, const wxString&, bool)':
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp:202:15: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
      || first != tem.rfind(key) // more than one instance
               ^
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp:203:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
      || !canEndWithKey && first == tem.length() - key.length())
                                 ^
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp:207:55: error: cannot pass objects of non-trivially-copyable type 'class wxCStrData' through '...'
                 name.c_str(), tem.c_str(), key.c_str());
                                                       ^
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp:207:55: error: cannot pass objects of non-trivially-copyable type 'class wxCStrData' through '...'
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp:207:55: error: cannot pass objects of non-trivially-copyable type 'class wxCStrData' through '...'
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp: In member function 'int GzippedFileReader::_ReadSync(void*, off_t, uint)':
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/CDVD/CompressedFileReader.cpp:442:13: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (first != maxInChunk)
             ^
pcsx2/CMakeFiles/pcsx2.dir/build.make:1885: recipe for target 'pcsx2/CMakeFiles/pcsx2.dir/CDVD/CompressedFileReader.cpp.o' failed
make[3]: *** [pcsx2/CMakeFiles/pcsx2.dir/CDVD/CompressedFileReader.cpp.o] Error 1
make[3]: Leaving directory '/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu'
CMakeFiles/Makefile2:303: recipe for target 'pcsx2/CMakeFiles/pcsx2.dir/all' failed
make[2]: *** [pcsx2/CMakeFiles/pcsx2.dir/all] Error 2
make[2]: Leaving directory '/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu'
Makefile:120: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu'
dh_auto_build: make -j1 returned exit code 2
debian/rules:53: recipe for target 'build' failed
make: *** [build] Error 2
```

I'm just saying that nothing will compile without using this cherry pick

https://github.com/PCSX2/pcsx2/commit/5933db386a886a2ee39f41cc84409c31b6110f50.patch

I'm not doing it because I want to. It's because I have to.

Using the same libraries and just switching the executable triggers this. This seems to be a sleeping/latent  that was triggered by that PR unless I'm missing something more obvious.

That is why I was thinking of backporting just that PR to see if old working code breaks too.

**Comment by avih** [comment]
Answer this then:

You have revision A, which after you cherry-pick 5933db386a886a2ee39f41cc84409c31b6110f50 - it works OK (no Vietnamese).

And you have the revision right after A, which after you cherry-pick 5933db386a886a2ee39f41cc84409c31b6110f50 - is broken (shows Vietnamese).

What's A? in a revision hash please.

**Comment by micove** [comment]
Revision A = 798fa11c57d5ebf26d0f88ca84c01180ea7fa2e0

**Comment by avih** [comment]
So 798fa11c57d5ebf26d0f88ca84c01180ea7fa2e0 + cherry-pick 5933db386a886a2ee39f41cc84409c31b6110f50  == working OK (no Vietnamese).

And 

806cefb0c60751b542255178e84aab859d917836 (which is one revision after 'A') + cherry-pick 5933db386a886a2ee39f41cc84409c31b6110f50  == broken (with Vietnamese)

?

**Comment by micove** [comment]
```
So 798fa11 + cherry-pick 5933db3 == working OK (no Vietnamese).
```
100% correct

```
806cefb (which is one revision after 'A') + cherry-pick 5933db3 == broken (with Vietnamese)
```
No, 1 revision after that is a11286bbc54a32e0a73163689243b7f3869b7c23
And as I tried to explain before too which is why you probably got confused

806cefb (which is one revision after 'A') + cherry-pick 5933db3  fails to build because
```
[ 54%] Building CXX object pcsx2/CMakeFiles/pcsx2.dir/gui/i18n.cpp.o
cd /tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu/pcsx2 && /usr/lib/ccache/c++   -DDOC_DIR_COMPILATION=/usr/share/doc/pcsx2 -DGAMEINDEX_DIR_COMPILATION=/usr/share/games/pcsx2 -DPLUGIN_DIR_COMPILATION=/usr/lib/i386-linux-gnu/pcsx2 -DWXUSINGDLL -DWX_PRECOMP -DXDG_STD -D_ARCH_32=1 -D_FILE_OFFSET_BITS=64 -D_M_X86=1 -D_M_X86_32=1 -D__WXGTK__ -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2  -msse -msse2 -march=i686 -pipe -std=c++0x -fvisibility=hidden -pthread -Wall -Wno-attributes -Wno-missing-field-initializers -Wno-unused-function -Wno-unused-parameter -Wno-unused-variable -Wno-unused-value  -Wstrict-aliasing -Wstrict-overflow=4  -D_FORTIFY_SOURCE=2  -Wformat -Wformat-security -DNDEBUG  -Wno-invalid-offsetof -pthread -I/usr/include/libdrm -I/usr/include/soundtouch -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/i386-linux-gnu/glib-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/lib/i386-linux-gnu/gtk-2.0/include -I/usr/include/SDL -isystem /usr/lib/i386-linux-gnu/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/common/include -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/common/include/Utilities -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/common/include/x86emitter -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu/common/include -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/gui -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/x86 -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu/pcsx2/gui -I/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/.    -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -fno-guess-branch-probability -fno-dse -fno-tree-dse -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -falign-functions -falign-jumps -falign-labels -falign-loops -fcaller-saves -fcprop-registers -fcrossjumping -fcse-follow-jumps -fcse-skip-blocks -fdefer-pop -fdelete-null-pointer-checks -fgcse -fgcse-lm -fif-conversion -fif-conversion2 -fmerge-constants -foptimize-sibling-calls -fpeephole2 -fregmove -freorder-blocks -freorder-functions -frerun-cse-after-loop -fsched-interblock -fsched-spec -fstrict-overflow -fthread-jumps -ftree-ccp -ftree-ch -ftree-copyrename -ftree-dce -ftree-dominator-opts -ftree-fre -ftree-lrs -ftree-pre -ftree-sra -ftree-ter -ftree-vrp -funit-at-a-time  -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -DWX_PRECOMP -fno-guess-branch-probability -fno-dse -fno-tree-dse -fno-omit-frame-pointer -fno-strict-aliasing -Wno-parentheses -Wstrict-aliasing -Wno-char-subscripts -Wno-missing-braces -DWX_PRECOMP -falign-functions -falign-jumps -falign-labels -falign-loops -fcaller-saves -fcprop-registers -fcrossjumping -fcse-follow-jumps -fcse-skip-blocks -fdefer-pop -fdelete-null-pointer-checks -fgcse -fgcse-lm -fif-conversion -fif-conversion2 -fmerge-constants -foptimize-sibling-calls -fpeephole2 -fregmove -freorder-blocks -freorder-functions -frerun-cse-after-loop -fsched-interblock -fsched-spec -fstrict-overflow -fthread-jumps -ftree-ccp -ftree-ch -ftree-copyrename -ftree-dce -ftree-dominator-opts -ftree-fre -ftree-lrs -ftree-pre -ftree-sra -ftree-ter -ftree-vrp -funit-at-a-time -DXDG_STD -o CMakeFiles/pcsx2.dir/gui/i18n.cpp.o -c /tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/gui/i18n.cpp
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/gui/i18n.cpp: In function 'wxLanguage i18n_FallbackToAnotherLang(wxLanguage)':
/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/pcsx2/gui/i18n.cpp:206:8: error: 'wxLANG_FRISIAN' was not declared in this scope
   case wxLANG_FRISIAN                  :
        ^
pcsx2/CMakeFiles/pcsx2.dir/build.make:3242: recipe for target 'pcsx2/CMakeFiles/pcsx2.dir/gui/i18n.cpp.o' failed
make[3]: *** [pcsx2/CMakeFiles/pcsx2.dir/gui/i18n.cpp.o] Error 1
make[3]: Leaving directory '/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu'
CMakeFiles/Makefile2:303: recipe for target 'pcsx2/CMakeFiles/pcsx2.dir/all' failed
make[2]: *** [pcsx2/CMakeFiles/pcsx2.dir/all] Error 2
make[2]: Leaving directory '/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu'
Makefile:120: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/tmp/buildd/pcsx2-1.2.1-836-ga11286b+dfsg/obj-i586-linux-gnu'
```

I had to do

a11286bbc54a32e0a73163689243b7f3869b7c23 (which is one revision after 'A') + cherry-pick 5933db3 + cherry-pick aea421cc8e0e45ac4d4eb55ea56d907aa2013ad1 == broken (with Vietnamese)

This is why I said that this was equivalent to 5933db386a886a2ee39f41cc84409c31b6110f50 since after those two required cherry-picks we are basically at the 1st commit I reported.

So basically 
a11286bbc54a32e0a73163689243b7f3869b7c23 is the culprit

**Comment by micove** [comment]
To make it even easier to understand here is a patch from the source that works vs the source that does not work.
```
diff -Naur pcsx2-1.2.1-834-g798fa11/build.sh pcsx2-1.2.1-839-g5933db3/build.sh
--- pcsx2-1.2.1-834-g798fa11/build.sh	2014-12-24 18:15:55.000000000 -0500
+++ pcsx2-1.2.1-839-g5933db3/build.sh	2014-12-29 07:25:07.000000000 -0500
@@ -16,14 +16,11 @@
 
 #set -e # This terminates the script in case of any error
 
-if [[ (-f /usr/bin/wx-config32-2.8 && -f /usr/bin/wxrc32-2.8) ]]; then
-#add flags for archlinux
-flags=(-DCMAKE_BUILD_PO=FALSE -DwxWidgets_CONFIG_EXECUTABLE='/usr/bin/wx-config32-2.8' -DwxWidgets_wxrc_EXECUTABLE='/usr/bin/wxrc32-2.8')
-else
 flags=(-DCMAKE_BUILD_PO=FALSE)
-fi
+
 cleanBuild=0
 useClang=0
+Build64=0
 
 for ARG in "$@"; do
     case "$ARG" in
@@ -73,6 +70,10 @@
     esac
 done
 
+if [[ (-f /usr/bin/wx-config32-2.8 && -f /usr/bin/wxrc32-2.8) && "$Build64" -eq 0 ]]; then
+#add flags for archlinux, wx 3 is not yet in main repositories, wx2.8 need to be used for now
+flags+=(-DwxWidgets_CONFIG_EXECUTABLE='/usr/bin/wx-config32-2.8' -DwxWidgets_wxrc_EXECUTABLE='/usr/bin/wxrc32-2.8' -DWX28_API=TRUE)
+fi
 root=$PWD/$(dirname "$0")
 log=$root/install_log.txt
 build=$root/build
diff -Naur pcsx2-1.2.1-834-g798fa11/pcsx2/gui/i18n.cpp pcsx2-1.2.1-839-g5933db3/pcsx2/gui/i18n.cpp
--- pcsx2-1.2.1-834-g798fa11/pcsx2/gui/i18n.cpp	2014-12-06 11:09:58.000000000 -0500
+++ pcsx2-1.2.1-839-g5933db3/pcsx2/gui/i18n.cpp	2015-01-03 09:52:43.000000000 -0500
@@ -196,33 +196,19 @@
 		case wxLANGUAGE_CHINESE_MACAU        : return wxLANGUAGE_CHINESE_TRADITIONAL;
 		case wxLANGUAGE_CHINESE_SINGAPORE    : return wxLANGUAGE_CHINESE_SIMPLIFIED;
 
-		// case wxLANGUAGE_SAMI_INARI_FINLAND    :
-		// case wxLANGUAGE_SAMI_NORTHERN_FINLAND :
-		// case wxLANGUAGE_SAMI_SKOLT_FINLAND    : return wxLANGUAGE_FINNISH;
-		// - not supported by wxWidgets (2.9.4).
-
-		case wxLANGUAGE_SAMI                 : 
-		// Most of the samis live in Norway.
-		// case wxLANGUAGE_SAMI_LULE_NORWAY     :
-		// case wxLANGUAGE_SAMI_NORTHERN_NORWAY :
-		// case wxLANGUAGE_SAMI_SOUTHERN_NORWAY :
-		// - not supported by wxWidgets (2.9.4).
-		case wxLANGUAGE_DANISH               :
+		// case wxLANGUAGE_SAMI                 : 
+		// case wxLANGUAGE_DANISH               :
 		case wxLANGUAGE_NORWEGIAN_NYNORSK    : return wxLANGUAGE_NORWEGIAN_BOKMAL;
 
-		//  case: wxLANGUAGE_SAMI_LULE_SWEDEN     :
-		//  case: wxLANGUAGE_SAMI_NORTHERN_SWEDEN :
-		//  case: wxLANGUAGE_SAMI_SOUTHERN_SWEDEN :
-		// - not supported by wxWidgets (2.9.4).
 		case wxLANGUAGE_SWEDISH_FINLAND      : return wxLANGUAGE_SWEDISH;
 
-		// case wxLANGUAGE_LUXEMBOURGISH        :
-		// - not supported by wxWidgets (2.9.4).
 		case wxLANGUAGE_AFRIKAANS            :
-		// case wxLANG_FRISIAN                  :
-		// - not supported by wxWidgets (2.9.4).
+		//case wxLANG_FRISIAN                  :
+		// case wxLANGUAGE_LUXEMBOURGISH        :
+		// - not supported by wxWidgets (3.0.0).
 		case wxLANGUAGE_DUTCH_BELGIAN        : return wxLANGUAGE_DUTCH;
 
+		case wxLANGUAGE_GALICIAN             :
 		case wxLANGUAGE_PORTUGUESE           : return wxLANGUAGE_PORTUGUESE_BRAZILIAN;
 
 		// Overkill 9000?
@@ -230,9 +216,9 @@
 		case wxLANGUAGE_GERMAN_BELGIUM       : 
 		case wxLANGUAGE_GERMAN_LIECHTENSTEIN : 
 		case wxLANGUAGE_GERMAN_LUXEMBOURG    : 
-		// Currently wxWidgets (2.9.4) doesn't support Sorbian.
-		//  case wxLANGUAGE_LOWER_SORBIAN :
-		//  case wxLANGUAGE_UPPER_SORBIAN :
+		// case wxLANGUAGE_LOWER_SORBIAN        :
+		// case wxLANGUAGE_UPPER_SORBIAN        :
+		// - Sorbian is not supported by wxWidgets (3.0.0).
 		case wxLANGUAGE_GERMAN_SWISS         : return wxLANGUAGE_GERMAN;
 
 		case wxLANGUAGE_BASQUE:
@@ -256,7 +242,7 @@
 		case wxLANGUAGE_SPANISH_URUGUAY:
 		case wxLANGUAGE_SPANISH_VENEZUELA: return wxLANGUAGE_SPANISH_MODERN;
 
-		case wxLANGUAGE_ITALIAN_SWISS        : return wxLANGUAGE_ITALIAN;
+		case wxLANGUAGE_ITALIAN_SWISS: return wxLANGUAGE_ITALIAN;
 
 		case wxLANGUAGE_CORSICAN:
 		case wxLANGUAGE_FRENCH_BELGIAN:
@@ -265,6 +251,12 @@
 		case wxLANGUAGE_FRENCH_MONACO:
 		case wxLANGUAGE_FRENCH_SWISS: return wxLANGUAGE_FRENCH;
 
+		case wxLANGUAGE_RUSSIAN_UKRAINE: return wxLANGUAGE_RUSSIAN;
+
+		case wxLANGUAGE_JAVANESE: return wxLANGUAGE_INDONESIAN;
+
+		case wxLANGUAGE_MALAY_BRUNEI_DARUSSALAM: return wxLANGUAGE_MALAY_MALAYSIA;
+
 		default                              : break;
 	}
 	return wxLangId;
```
It even includes the hashsum in the folder name.

**Comment by micove** [comment]
Sigh ... apparently I compiled the broken one with wx2.8 since I was testing a different bug which only occurs with wx2.8. Then when I started recompiling earlier I was using w3.0. That is why I could never find the issue since I recompiled all those versions using wx3.0.

Back to square 1.

**Comment by gregory38** [comment]
wx3.0 (only keep alphanumerical char) sorts array differently than wx2.8.

For example I found a bug in First Time Wizard because " Default" wasn't the first item of the array (notice extra space). Arabia was selected instead.
I'm nearly issue it is a similar issue. Vietnamese is probably latest item of the language array. Position is likely wrong somewhere.

**Comment by avih** [comment]
Gregory, but he says that the first time wizard doesn't have this issue - that is, if I understood him correctly. Maybe he should confirm this before you act according to this info.

**Comment by gregory38** [comment]
I fixed FTW recently. I never tested yet the new way to change language.

However it is not clear for me, is it a 2.8 or 3.0 issue?

**Comment by avih** [comment]
Not clear to me too.

**Comment by micove** [comment]
>Sigh ... apparently I compiled the broken one with wx2.8 since I was testing a different bug which only occurs with wx2.8.

**Comment by micove** [comment]
Using wx3.0, it never triggered.
Using wx2.8, just compiled.

c4bd50c (Vietnamese)
1779ee4 (No Vietnamese, well there is no new window)

In both cases it never shows in the 1st time configuration window.

Gregory might be onto something since:
https://www.wxwidgets.org/about/translations/

Sorted by language, Vietnamese is last and looking at
http://svn.wxwidgets.org/svn/wx/wxWidgets/trunk/locale/vi.po

That is all the text that gets changed.

**Comment by gregory38** [comment]
no it is not related.
```
22:02 gregory ~/playstation/emulateur/pcsx2_merge (3kinox-64-bit-clean|✚1…)% dpkg -S /usr/share/locale/vi/LC_MESSAGES/libcwidget3.mo  
libcwidget3:amd64: /usr/share/locale/vi/LC_MESSAGES/libcwidget3.mo
22:02 gregory ~/playstation/emulateur/pcsx2_merge (3kinox-64-bit-clean|✚1…)% sho libcwidget3
Description-en: high-level terminal interface library for C++ (runtime files)
 libcwidget is a modern user interface library modeled on GTK+ and Qt,
 but using curses as its display layer and with widgets that are
 tailored to a terminal environment.
 .
 This package contains the files that are required to run programs
 compiled against libcwidget.
Description-md5: 25817f3f038c1f33b9d63aafe32c6d75
Description-md5: 25817f3f038c1f33b9d63aafe32c6d75
Multi-Arch: same
```

**Comment by micove** [comment]
I removed all the mo files one by one and ran the program each time.
I left my end guess for last

```
$ find /usr/share/locale/ -type f | xargs grep "Thôi"
Binary file /usr/share/locale/vi/LC_MESSAGES/gtk20.mo matches
```
the Vietnamese still happens

```
$ sudo rm -f /usr/share/locale/vi/LC_MESSAGES/gtk20.mo
$ find /usr/share/locale/ -type f | xargs grep "Thôi"
$ pcsx2
```

Victory.

**Comment by micove** [comment]
To confirm it's truly that one

I restored all the mo files
```
$ sudo cp -R vi/ /usr/share/locale/
```

```
$ sudo rm -f /usr/share/locale/vi/LC_MESSAGES/gtk20.mo
$ find /usr/share/locale/ -type f | xargs grep "Thôi"
Binary file /usr/share/locale/vi/LC_MESSAGES/libcwidget3.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/newt.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/mate-session-manager.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/gtk30-properties.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/gnome-vfs-2.0.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/marco.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/gtk20-properties.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/zenity.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/gvfs.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/gcr.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/caja.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/nautilus.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/xchat.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/aptitude.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/gtk30.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/nano.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/GConf2.mo matches
Binary file /usr/share/locale/vi/LC_MESSAGES/libgphoto2-6.mo matches
$ pcsx2
```

Then the Vietnamese is gone by just removing gtk20.mo

**Comment by gregory38** [comment]
Ok. I have the mo file. Do you see in the log a loading database message?

**Comment by micove** [comment]
No, I enabled all the logging options and 0 message appear in the console or the "Program Log"

I did a strace pcsx2
```
micove@debian:~/Desktop$ cat strace.txt | grep "gtk20\.mo"
open("/usr/share/locale/en_US.UTF-8/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en_US.utf8/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en_US/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en.UTF-8/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en.utf8/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/vi_VN/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/vi/LC_MESSAGES/gtk20.mo", O_RDONLY) = 6
```

best thing I could find from the huge output

**Comment by micove** [comment]
I uploaded the output of 
```
cat strace.txt | grep "/usr/share/locale"
```
http://paste.debian.net/hidden/5246319f/

**Comment by gregory38** [comment]
Ok. I will give check the behavior of my computer.

**Comment by micove** [comment]
It seems to try en_US first which is my default then can't find it since it's not here
https://packages.debian.org/sid/all/libgtk2.0-common/filelist
and then tries all the languages  supported by wx2.8 skipping a ton of the ones not supported. And just defaults to the last one from wx2.8 which is vi.

It seems that you need to have en_US or a locale not in
https://packages.debian.org/sid/all/libgtk2.0-common/filelist
 to trigger it but not sure about this requirement.

**Comment by gregory38** [comment]
Hum ?????
```
0:10 gregory ~/playstation/emulateur/pcsx2_merge (master|…)% grep "gtk20" /tmp/strace.txt 
open("/usr/share/locale/en_US/LC_MESSAGES/gtk20-properties.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/gtk20-properties.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en_US/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/gtk20.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
```

```0:10 gregory ~/playstation/emulateur/pcsx2_merge (master|…)% grep "/usr/share/locale/vi" /tmp/strace.txt                             
stat64("/usr/share/locale/vi", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat64("/usr/share/locale/vi/pcsx2_Main.mo", 0xffa8168c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi/LC_MESSAGES/pcsx2_Main.mo", 0xffa8168c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat64("/usr/share/locale/vi/pcsx2_Main.mo", 0xffa8168c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi/LC_MESSAGES/pcsx2_Main.mo", 0xffa8168c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi_VN.UTF-8/LC_MESSAGES/pcsx2_Main.mo", 0xffa8180c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi_VN.UTF-8/pcsx2_Main.mo", 0xffa8180c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi_VN/LC_MESSAGES/pcsx2_Main.mo", 0xffa8180c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi_VN/pcsx2_Main.mo", 0xffa8180c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi/LC_MESSAGES/pcsx2_Main.mo", 0xffa8180c) = -1 ENOENT (No such file or directory)
stat64("/usr/share/locale/vi/pcsx2_Main.mo", 0xffa8180c) = -1 ENOENT (No such file or directory)

**Comment by micove** [comment]
What distro are you running?

I just did some quick test:
Debian sid amd64 => Viet
Debian jessie i386 => Works
Arch Linux amd64 with multilib => Works
Fedora amd64 => Viet even in the 1st time configuration wizard which i did not notice but it happens on Debian/sid too.

Fedora:
![screenshot-fedora running - oracle vm virtualbox](https://cloud.githubusercontent.com/assets/1458001/5623546/de2b4846-9523-11e4-844f-3e0d294b143a.png)

Debian but also happens in Fedora (I missed it before)
![screenshot-pcsx2 first time configuration](https://cloud.githubusercontent.com/assets/1458001/5623562/14437fe8-9524-11e4-9f05-dbbe7fd0ee66.png)


The Debian/jessie, Fedora, Arch had fairly recent clean install and really default settings. Well the arch experiences requires hand tuning but it was fairly standard.

Thanks to the 2nd picture which I did not notice before I discovered that this is an old issue since it also happens in revision prior to the new window appearing. The difference is that the Vietnamese stays just in the 1st configuration wizard and never propagates into the main window which is probably why I never noticed. With the new window it propagates in the main window and it's way obvious to notice.

It could be some dependency version issue in which a new library version breaks stuff since it works in Debian/testing but not Debian/sid, in Arch but not in Fedora.  It seems to be fairly recent too since I don't remember seeing it before but I was using Debian/stable until a few weeks ago.

Might as well try it in a clean Debian/sid installation in a virtual machine.

**Comment by avih** [comment]
The Debian plugins-page plugins-search-path of the FTW is probably a bug and should be fixed, especially since other parts of this dialog are displayed correctly.

The change language dialog can be "fixed" by adding an extra argument to the language selection panel - the language to start at, or it should use the one which is already set. Though the latter could be non trivial since the same dialog also appears at the FTW - at which case the language to choose is less clear, and it might even be hard to tell if a specific language is set at all. But maybe not, needs to look at the code.

What I don't get though, is how come on some systems there's a difference between the default language at the FTW vs the default language at the change-language dialog. This is probably something we should look into.

**Comment by micove** [comment]
Okay, I found the package that triggers the bug.

I did a clean install of Debian/testing amd64=>. Works
Upgraded to sid=> Works

No Vietnamese so I was happy since for a second but when I clicked at the language dialogue list only had:
"System Default"
"English" 

That means that I had not installed the locales-all package with all the pre-compiled locales. 

So as soon as I just installed "locales-all" the list contained all the available languages as expected but the automatic Vietnamese thing returned.

I can confirm that removing that package fixes it for my system and all other Debian cases. While installing it breaks all. 

Since the archlinux way is doing everything manually I only compiled en_US-utf8 so that is probably why it worked. I just checked and only "System Default" and "English" appear. Fedora goes a bit overboard and that is why it does not work:
![screenshot-fedora running - oracle vm virtualbox](https://cloud.githubusercontent.com/assets/1458001/5624776/7c6c6632-953a-11e4-9485-add46c00a3b3.png)


My original strace had PCSX2 checking each and every file in here
https://packages.debian.org/sid/amd64/locales-all/filelist
in the /usr/lib/locales

**Comment by gregory38** [comment]
Hum could you try a standard portable build (not package mode) ?

**Comment by micove** [comment]
Sure, I compiled the current HEAD 1.2.1-877-gc5e6013

```
$ sudo apt-get install pcsx2 libwxgtk2.8-0:i386 libwxbase2.8-0:i386  locales-all
$ sudo apt-get remove --purge pcsx2
========
In a chroot:
# ./build.sh --release --no-simd --wx28 --glsl -DREBUILD_SHADER=TRUE
========
Move the bin folder out of the chroot and:
$ cd bin/
$ ./pcsx2
```
Same as before, still Vietnamese.

**Comment by gregory38** [comment]
Let's try something
Fire PCSX2 in gdb, run. Then add a breakpoint on open (break open). Continue. Click on the language selector. Hopefully it would break on something, if you can send me the backtrace (bt). You might need to add debug symbol of wx/gtk/stdc lib.

**Comment by micove** [comment]
Using the current HEAD  1.3.1-1-gdacadcd. and debian packaging.

```
sudo apt-get install pcsx2 pcsx2-dbg gdb libwxgtk2.8-dbg:i386 libwxbase2.8-dbg:i386 libgtk2.0-0-dbg:i386 libc6-dbg:i386 locales-all
```

Not sure if this is right or helpful since nothing breaks when using the language selector on the First Configuration wizard. It broke when clicking next but that backtrace does not look helpful. Also open breaks 105 times so I skipped the 1st 100.
```
$ gdb pcsx2
(gdb) break open
Function "open" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (open) pending.
(gdb) ignore 1 100
Will ignore next 100 crossings of breakpoint 1.
(gdb) run
Starting program: /usr/games/pcsx2 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Interface is initializing.  Entering Pcsx2App::OnInit!
Applying operating system default language...
Command line parsing...
Command line parsed!
[New Thread 0xf47bdb40 (LWP 23782)]

Breakpoint 1, open () at ../sysdeps/unix/syscall-template.S:81
81	../sysdeps/unix/syscall-template.S: No such file or directory.
(gdb) continue
Continuing.

Breakpoint 1, open () at ../sysdeps/unix/syscall-template.S:81
81	in ../sysdeps/unix/syscall-template.S
(gdb) continue
Continuing.
[New Thread 0xf0642b40 (LWP 23803)]

Breakpoint 1, open () at ../sysdeps/unix/syscall-template.S:81
81	in ../sysdeps/unix/syscall-template.S
(gdb) continue
Continuing.
[New Thread 0xefce6b40 (LWP 23804)]
[New Thread 0xef4e5b40 (LWP 23805)]
[New Thread 0xeece4b40 (LWP 23806)]

Breakpoint 1, open () at ../sysdeps/unix/syscall-template.S:81
81	../sysdeps/unix/syscall-template.S: No such file or directory.
(gdb) continue
Continuing.
[Thread 0xefce6b40 (LWP 23804) exited]
[Thread 0xef4e5b40 (LWP 23805) exited]

(pcsx2:23771): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 15
[New Thread 0xef4e5b40 (LWP 23817)]
[New Thread 0xefce6b40 (LWP 23818)]
[New Thread 0xee2ffb40 (LWP 23819)]

(pcsx2:23771): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 15
[Thread 0xee2ffb40 (LWP 23819) exited]
[Thread 0xef4e5b40 (LWP 23817) exited]
[Thread 0xeece4b40 (LWP 23806) exited]
[Thread 0xefce6b40 (LWP 23818) exited]

(pcsx2:23771): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 15

(pcsx2:23771): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 15

(pcsx2:23771): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 15

Breakpoint 1, open () at ../sysdeps/unix/syscall-template.S:81
81	../sysdeps/unix/syscall-template.S: No such file or directory.
(gdb) info breakpoints
Num     Type           Disp Enb Address    What
1       breakpoint     keep y   <MULTIPLE> 
	breakpoint already hit 105 times
1.1                         y     0xf7027960 ../sysdeps/unix/syscall-template.S:81
1.2                         y     0xf71071d0 ../sysdeps/unix/syscall-template.S:81
(gdb) bt
#0  open () at ../sysdeps/unix/syscall-template.S:81
#1  0xf6fbd19c in __GI__IO_file_open (fp=fp@entry=0xa2701f0, 
    filename=filename@entry=0xa26a550 "/home/micove/.config/PCSX2/logs/emuLog.txt", 
    posix_mode=posix_mode@entry=577, prot=prot@entry=438, read_write=read_write@entry=4, 
    is32not64=is32not64@entry=0) at fileops.c:227
#2  0xf6fbd34b in _IO_new_file_fopen (fp=fp@entry=0xa2701f0, 
    filename=filename@entry=0xa26a550 "/home/micove/.config/PCSX2/logs/emuLog.txt", mode=<optimized out>, 
    mode@entry=0x8373afd "wb", is32not64=is32not64@entry=0) at fileops.c:332
#3  0xf6fb1cab in __fopen_internal (
    filename=filename@entry=0xa26a550 "/home/micove/.config/PCSX2/logs/emuLog.txt", 
    mode=mode@entry=0x8373afd "wb", is32=is32@entry=0) at iofopen.c:90
#4  0xf6fb41db in _IO_fopen64 (filename=0xa26a550 "/home/micove/.config/PCSX2/logs/emuLog.txt", 
    mode=mode@entry=0x8373afd "wb") at iofopen64.c:39
#5  0x08173334 in RelocateLogfile () at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppConfig.cpp:1067
#6  0x0819061d in AppApplySettings (oldconf=oldconf@entry=0xffffb6d8)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppMain.cpp:751
#7  0x081a8046 in ApplyStateStruct::ApplyPage (this=this@entry=0xa09a240, pageid=pageid@entry=-1)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/Panels/BaseApplicableConfigPanel.cpp:92
#8  0x081a8f2a in ApplyStateStruct::ApplyAll (this=0xa09a240)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/Panels/BaseApplicableConfigPanel.cpp:128
#9  0x081b6273 in FirstTimeWizard::OnPageChanging (this=0xffffd01c, evt=...)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/Dialogs/FirstTimeWizard.cpp:214
#10 0x0818f8b6 in Pcsx2App::HandleEvent (this=0x9f9f248, handler=0xffffd01c, func=<optimized out>, event=...)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppMain.cpp:552
#11 0xf7f3d5e8 in wxEvtHandler::ProcessEventIfMatches(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&) ()
   from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#12 0xf7f3d9c3 in wxEvtHandler::SearchDynamicEventTable(wxEvent&) ()
   from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#13 0xf7f3da9e in wxEvtHandler::ProcessEvent(wxEvent&) () from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#14 0xf7d0f8c9 in wxWindowBase::TryParent(wxEvent&) () from /usr/lib/i386-linux-gnu/libwx_gtk2u_core-2.8.so.0
#15 0xf7f3da58 in wxEvtHandler::ProcessEvent(wxEvent&) () from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#16 0xf7a8862f in wxWizard::ShowPage(wxWizardPage*, bool) () from /usr/lib/i386-linux-gnu/libwx_gtk2u_adv-2.8.so.0
#17 0xf7a8938d in wxWizard::OnBackOrNext(wxCommandEvent&) () from /usr/lib/i386-linux-gnu/libwx_gtk2u_adv-2.8.so.0
#18 0x0818f8b6 in Pcsx2App::HandleEvent (this=0x9f9f248, handler=0xffffd01c, func=<optimized out>, event=...)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppMain.cpp:552
#19 0xf7f3d5e8 in wxEvtHandler::ProcessEventIfMatches(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&) ()
   from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#20 0xf7f3d6b6 in wxEventHashTable::HandleEvent(wxEvent&, wxEvtHandler*) ()
   from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
---Type <return> to continue, or q <return> to quit---
#21 0xf7f3dac5 in wxEvtHandler::ProcessEvent(wxEvent&) () from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#22 0xf7d0f8c9 in wxWindowBase::TryParent(wxEvent&) () from /usr/lib/i386-linux-gnu/libwx_gtk2u_core-2.8.so.0
#23 0xf7f3da58 in wxEvtHandler::ProcessEvent(wxEvent&) () from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#24 0xf7c46b46 in ?? () from /usr/lib/i386-linux-gnu/libwx_gtk2u_core-2.8.so.0
#25 0xf79a64e7 in g_cclosure_marshal_VOID__VOIDv () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#26 0xf79a4a5f in ?? () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#27 0xf79be7f9 in g_signal_emit_valist () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#28 0xf79bf0d5 in g_signal_emit () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#29 0xf730a3c1 in IA__gtk_button_clicked (button=0xa026438)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkbutton.c:1115
#30 0xf730b571 in gtk_real_button_released (button=0xa026438)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkbutton.c:1712
#31 0xf79a64c7 in g_cclosure_marshal_VOID__VOIDv () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#32 0xf79a32e2 in ?? () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#33 0xf79a4a5f in ?? () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#34 0xf79be7f9 in g_signal_emit_valist () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#35 0xf79bf0d5 in g_signal_emit () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#36 0xf730a2f1 in gtk_button_released (button=0xa026438)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkbutton.c:1107
#37 0xf730a334 in gtk_button_button_release (widget=0xa026438, event=0xa26d940)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkbutton.c:1604
#38 0xf73b7bac in _gtk_marshal_BOOLEAN__BOXED (closure=0x9ff0078, return_value=0xffffcb3c, n_param_values=2, 
    param_values=0xffffcba0, invocation_hint=0xffffcb58, marshal_data=0xf730a300 <gtk_button_button_release>)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkmarshalers.c:86
#39 0xf79a3394 in ?? () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#40 0xf79a483b in g_closure_invoke () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#41 0xf79b6751 in ?? () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#42 0xf79bea52 in g_signal_emit_valist () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#43 0xf79bf0d5 in g_signal_emit () from /usr/lib/i386-linux-gnu/libgobject-2.0.so.0
#44 0xf74d88fc in gtk_widget_event_internal (widget=0xa026438, event=0xa26d940)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkwidget.c:5010
#45 0xf74d8bdc in IA__gtk_widget_event (widget=0xa026438, event=0xa26d940)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkwidget.c:4807
#46 0xf73b5f49 in IA__gtk_propagate_event (widget=0xa026438, event=0xa26d940)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkmain.c:2490
#47 0xf73b645d in IA__gtk_main_do_event (event=0xa26d940)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkmain.c:1685
#48 0xf77e1189 in gdk_event_dispatch (source=0x9fc7900, callback=0x0, user_data=0x0)
    at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gdk/x11/gdkevents-x11.c:2403
---Type <return> to continue, or q <return> to quit---
#49 0xf6affda4 in g_main_context_dispatch () from /lib/i386-linux-gnu/libglib-2.0.so.0
#50 0xf6b000c9 in ?? () from /lib/i386-linux-gnu/libglib-2.0.so.0
#51 0xf6b00479 in g_main_loop_run () from /lib/i386-linux-gnu/libglib-2.0.so.0
#52 0xf73b5135 in IA__gtk_main () at /build/gtk+2.0-0IZDpU/gtk+2.0-2.24.25/gtk/gtkmain.c:1257
#53 0xf7bf9c23 in wxEventLoop::Run() () from /usr/lib/i386-linux-gnu/libwx_gtk2u_core-2.8.so.0
#54 0xf7c574df in wxDialog::ShowModal() () from /usr/lib/i386-linux-gnu/libwx_gtk2u_core-2.8.so.0
#55 0xf7a8925c in wxWizard::RunWizard(wxWizardPage*) () from /usr/lib/i386-linux-gnu/libwx_gtk2u_adv-2.8.so.0
#56 0x08185b5a in DoFirstTimeWizard () at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppUserMode.cpp:224
#57 0x08185e68 in Pcsx2App::EstablishAppUserMode (this=this@entry=0x9f9f248)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppUserMode.cpp:304
#58 0x081884e9 in Pcsx2App::DetectCpuAndUserMode (this=this@entry=0x9f9f248)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppInit.cpp:53
#59 0x0818b202 in Pcsx2App::OnInit (this=0x9f9f248)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppInit.cpp:447
#60 0xf7edab1e in wxEntry(int&, wchar_t**) () from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#61 0xf7edabc3 in wxEntry(int&, char**) () from /usr/lib/i386-linux-gnu/libwx_baseu-2.8.so.0
#62 0x0818d9bb in main (argc=1, argv=0xffffd484)
    at /tmp/buildd/pcsx2-1.3.1-1-gdacadcd+dfsg/pcsx2/gui/AppMain.cpp:53
(gdb)
```

Well while doing this I noticed this.
```
Action canceled: Startup aborted: User canceled FirstTime Wizard.
[wx] Failed to create a temporary file name (error 2: Không có tập tin hoặc thư mục như vậy)
[wx] can't open user configuration file.
[wx] Failed to create a temporary file name (error 2: Không có tập tin hoặc thư mục như vậy)
[wx] can't open user configuration file.
[wx] Failed to create a temporary file name (error 2: Không có tập tin hoặc thư mục như vậy)
[wx] can't open user configuration file.
```
I just ran pcsx2 with no config folder and hit cancel in the 1st configuration wizard. Since I just ran the program and hit cancel (nothing else was done) and Vietnamese appears in the error message then it's safe to assume that whatever gets loaded happens before the first configuration window. I also tried with the new language selection window but nothing broke before the Vietnamese appeared.

**Comment by gregory38** [comment]
```
Nothing really useful in those bt.
```
Yep.

By the way, what is your desktop, do you use special gtk theme or whatever?

**Comment by micove** [comment]
Nothing special.
The default one from Mate which is Menta.
In Fedora can't really remember but whatever they use for default in LXDE.

**Comment by gregory38** [comment]
Thanks for all the tests. As I don't have time to install  vm currently (neither disk space), I will mark this fix as low priority. Besides I'm confident that the WX3.0 transition will be fast enough.

**Comment by bositman** [comment]
This seems fixed, reopen if anyone thinks it's still there.
